### ISSUE TITLE
[FEA] Commutative bitwise operators (e.g. __rxor__) for cudf.Series.

### ISSUE THREAD

**Issue by bdice** [rapidsai/cudf#10177]
**Is your feature request related to a problem? Please describe.**
`cudf.Series` does not have right-side binary operators for bitwise operators like `__rxor__`. This was caught after changing the output of `Series.hash_values` from a `cupy.array` to a `cudf.Series` in #9390, which broke NVTabular: https://github.com/NVIDIA-Merlin/NVTabular/pull/1376#pullrequestreview-868327738. The `cupy.array` class supports both directions, like `array ^ scalar` and `scalar ^ array`. Currently `cudf.Series` only supports `series ^ scalar`.

**Describe the solution you'd like**
Add reverse binary operators for all cases that are missing, which include at least bitwise operators like `__rxor__` but possibly others as well. This kind of problem has been addressed several times in the past for other operators (#208, #213, #8598, ...). Resolving this issue should involve checking the full list in the [Python object model documentation](https://docs.python.org/3/reference/datamodel.html#object.__radd__) to make sure we catch everything that pandas supports.

**Additional context**
A workaround is to use the binary operator in the reverse order (for commutative operators).

**Comment by bdice** [comment]
List of other operators I've found are implemented in pandas but not cudf:
- `__divmod__(self, other)`
- `__rdivmod__(self, other)`
- `__round__(self, ndigits)`

**Comment by vyasr** [comment]
#10360 handles the reverse binary operators. The three that you listed in your last comment are not yet done. divmod and rdivmod are out of scope for #10360 but are in general trivial to implement in the relevant classes once that PR is merged since they are just 
```
def __divmod__(self, other):
    return self // other, self % other
def __rdivmod__(self, other):
    return other // self, other % self
```
I think they should probably just bypass the new binop machinery, but maybe there will be a benefit to including them.

`round` is a little different from other binary operations because the second parameter is optional, i.e. `round(x)` will not pass a parameter for `ndigits` so you have to make sure to handle the `None` case. Obviously trivial to do, but a reason to perhaps not shoehorn it into the same patterns and instead implement it separately.

**Comment by bdice** [comment]
@vyasr Awesome. I subscribed to notifications on #10360 and I can take a look at the missing operators after that PR is merged. I changed the title of this issue to reflect the change in scope.
